### ISSUE TITLE
Crash on wpcom login LoginActivity

### ISSUE THREAD

**Issue by mzorz** [wordpress-mobile/WordPress-Android#7147]
Crash happened on emulator, Android 7.0.0, maybe similar to #3209 regarding how the Dialog was being used?

```
01-22 07:16:17.312 1750-1750/org.wordpress.android I/WordPress-NOTIFS: notifications update service > completed
01-22 07:16:17.408 1750-1750/org.wordpress.android I/WordPress-NOTIFS: notifications update service > destroyed
01-22 07:16:17.431 1750-1750/org.wordpress.android D/WordPress-READER: reader post list > saving instance state
01-22 07:16:17.479 1750-1750/org.wordpress.android E/WindowManager: android.view.WindowLeaked: Activity org.wordpress.android.ui.accounts.LoginActivity has leaked window DecorView@859b980[] that was originally added here
                                                                        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:417)
                                                                        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:331)
                                                                        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:93)
                                                                        at android.app.Dialog.show(Dialog.java:316)
                                                                        at android.app.ProgressDialog.show(ProgressDialog.java:116)
                                                                        at org.wordpress.android.ui.accounts.login.LoginBaseFormFragment.startProgress(LoginBaseFormFragment.java:221)
                                                                        at org.wordpress.android.ui.accounts.login.LoginBaseFormFragment.startProgress(LoginBaseFormFragment.java:213)
                                                                        at org.wordpress.android.ui.accounts.login.Login2FaFragment.doAuthAction(Login2FaFragment.java:278)
                                                                        at org.wordpress.android.ui.accounts.login.Login2FaFragment.next(Login2FaFragment.java:269)
                                                                        at org.wordpress.android.ui.accounts.login.Login2FaFragment.OnEditorCommit(Login2FaFragment.java:338)
                                                                        at org.wordpress.android.widgets.WPLoginInputRow$1.onEditorAction(WPLoginInputRow.java:186)
                                                                        at android.widget.TextView.onKeyUp(TextView.java:6385)
                                                                        at android.view.KeyEvent.dispatch(KeyEvent.java:2699)
                                                                        at android.view.View.dispatchKeyEvent(View.java:9876)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.widget.ScrollView.dispatchKeyEvent(ScrollView.java:379)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at android.view.ViewGroup.dispatchKeyEvent(ViewGroup.java:1667)
                                                                        at com.android.internal.policy.DecorView.superDispatchKeyEvent(DecorView.java:403)
                                                                        at com.android.internal.policy.PhoneWindow.superDispatchKeyEvent(PhoneWindow.java:1800)
                                                                        at android.app.Activity.dispatchKeyEvent(Activity.java:3021)
                                                                        at android.support.v7.app.AppCompatActivity.dispatchKeyEvent(AppCompatActivity.java:534)
                                                                        at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
                                                                        at android.support.v7.app.AppCompatDelegateImplBase$AppCompatWindowCallbackBase.dispatchKeyEvent(AppCompatDelegateImplBase.java:316)
                                                                        at android.support.v7.view.WindowCallbackWrapper.dispatchKeyEvent(WindowCallbackWrapper.java:58)
                                                                        at com.android.internal.policy.DecorView.dispatchKeyEvent(DecorView.java:317)
                                                                        at android.view.ViewRootImpl$ViewPostImeInputStage.processKeyEvent(ViewRootImpl.java:4327)
                                                                        at android.view.ViewRootImpl$ViewPostImeInputStage.onProcess(ViewRootImpl.java:4298)
                                                                        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3849)
                                                                        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3902)
                                                                        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3868)
                                                                        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:3995)
                                                                        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3876)
                                                                        at android.view.ViewRootImpl$AsyncInputStage.apply(ViewRootImpl.java:4052)
                                                                        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3849)
                                                                        at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3902)
                                                                        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3868)
                                                                        at android.view.ViewRootImpl$InputStage.apply(ViewRootImpl.java:3876)
                                                                        at android.view.ViewRootImpl$InputStage.deliver(ViewRootImpl.java:3849)
01-22 07:16:17.479 1750-1750/org.wordpress.android E/WindowManager:     at android.view.ViewRootImpl$InputStage.onDeliverToNext(ViewRootImpl.java:3902)
                                                                        at android.view.ViewRootImpl$InputStage.forward(ViewRootImpl.java:3868)
                                                                        at android.view.ViewRootImpl$AsyncInputStage.forward(ViewRootImpl.java:4028)
                                                                        at android.view.ViewRootImpl$ImeInputStage.onFinishedInputEvent(ViewRootImpl.java:4189)
                                                                        at android.view.inputmethod.InputMethodManager$PendingEvent.run(InputMethodManager.java:2365)
                                                                        at android.view.inputmethod.InputMethodManager.invokeFinishedInputEventCallback(InputMethodManager.java:1961)
                                                                        at android.view.inputmethod.InputMethodManager.finishedInputEvent(InputMethodManager.java:1952)
                                                                        at android.view.inputmethod.InputMethodManager$ImeInputEventSender.onInputEventFinished(InputMethodManager.java:2342)
                                                                        at android.view.InputEventSender.dispatchInputEventFinished(InputEventSender.java:141)
                                                                        at android.os.MessageQueue.nativePollOnce(Native Method)
                                                                        at android.os.MessageQueue.next(MessageQueue.java:323)
                                                                        at android.os.Looper.loop(Looper.java:136)
                                                                        at android.app.ActivityThread.main(ActivityThread.java:6077)
                                                                        at java.lang.reflect.Method.invoke(Native Method)
                                                                        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:865)
                                                                        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:755)
01-22 07:16:17.486 642-1121/system_process I/WindowManager: Destroying surface Surface(name=org.wordpress.android/org.wordpress.android.ui.accounts.LoginActivity) called by com.android.server.wm.WindowStateAnimator.destroySurface:2014 com.android.server.wm.WindowStateAnimator.destroySurfaceLocked:881 com.android.server.wm.WindowState.removeLocked:1449 com.android.server.wm.WindowManagerService.removeWindowInnerLocked:2478 com.android.server.wm.WindowManagerService.removeWindowLocked:2436 com.android.server.wm.WindowManagerService.removeWindowLocked:2305 com.android.server.wm.AppWindowToken.removeAllWindows:530 com.android.server.wm.AppWindowToken.removeAppFromTaskLocked:326 
```

**Comment by mzorz** [comment]
Steps to repro:
Using `PREVIEW - Google Nexus 5X - 7.0.0` genymotion image
`vanillaRelease` apk
install from scratch (or uninstall and install)
1. make sure to have 2fa enabled
2. start the app, enter your email address
3. choose "enter password instead" link
4. enter password
5. when prompted, enter 2fa code
6. see the progress dialog either be black (or the full screen be black as happened to me first time), and unresponsive for a few seconds, and see the log appear in the logcat

<img width="695" alt="screen shot 2018-01-22 at 9 45 11 am" src="https://user-images.githubusercontent.com/6597771/35221637-a47a9b26-ff59-11e7-9995-75c12f994173.png">

**Comment by hypest** [comment]
Couldn't reproduce it on Genymotion v2.11.0, VirtualBox v5.2.6 r120293 (Qt5.6.3), virtual device Nexus 5X running 7.0.0 - API 24 - 1080x1920.

This highly seems like an issue with Genymotion and the particular image reported on @mzorz . The image seems to be unable to reliably render the progress dialog.

Regarding the "unresponsiveness", that's expected under that flow since the operation is non-interruptible (final phases of logging in), especially if there are many sites to be fetched for the account. I suspect it only looks like "freeze" because the progress dialog is not rendering.

In the end, the logcat message is not causing a problem _during login_ or after anyway. It's a `WindowLeaked` warning (even though it's logged as `E/` error) that happens because we don't dismiss the progress dialog before finishing the Activity (`LoginActivity.java`). It doesn't cause a crash or freeze at any time though.

I will prepare a PR with a fix for the WindowLeaked issue so we reduce the noise from it.

**Comment by hypest** [comment]
I opened #7154 with a fix for the logged warning. @mzorz when you have the chance, can you please try it on the Genymotion image and see if the logging message is still occurring? The "black" progress bar itself should still happen on your image since the PR doesn't try to address it. Thanks!

**Comment by mzorz** [comment]
>  I suspect it only looks like "freeze" because the progress dialog is not rendering.

Sounds quite plausible!

**Comment by mzorz** [comment]
Closed via #7154
